qemu/tests/test_control_virtqueue.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_net
+from virttest.utils_test.qemu import MemoryHotplugTest
+
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class MemoryHotplugRepeat(MemoryHotplugTest):
+
+    def hotplug_mem(self, vm, target_mem):
+        """
+        Hotplug memory in target_mem
+        :param vm: vm object in this test
+        :param target_mem: target memory to be hotplugged
+        """
+        self.hotplug_memory(vm, target_mem)
+        self.check_memory(vm)
+
+    def start_test(self):
+        """
+        Prepare required test params, hotplug 1 memory to update device
+        """
+        vm = self.env.get_vm(self.params["main_vm"])
+        session_serial = vm.wait_for_serial_login()
+        target_mem = self.params["target_mem"]
+        error_context.context("Hotplug memory %s" % target_mem, LOG_JOB.info)
+        self.hotplug_memory(vm, target_mem)
+        vm.verify_kernel_crash()
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test vDPA control virtqueue
+
+    1) Boot a guest and qemu must have space for more memory slots
+    2) Change mac address 2^16-1 times
+    3) Hotplug 1G memory to update device
+    4) Try to update the mac again and it should mac address can be changed
+       normally
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = int(params.get("login_timeout", 360))
+    session_serial = vm.wait_for_serial_login(timeout=timeout)
+    size_mem = params.get("size_mem")
+    change_cmd = params["change_cmd"]
+    old_mac = vm.get_mac_address(0)
+    new_mac = vm.virtnet.generate_mac_address(0)
+    interface = utils_net.get_linux_ifname(session_serial, old_mac)
+    change_times = params.get_numeric("change_times")
+    for i in range(change_times):
+        output = session_serial.cmd_output_safe(change_cmd % (new_mac,
```

Comment:
Where is the `output` used here?

Reply:
I would like to used this output to judge if it can changed mac address normally, but forgot to add the related steps. I will add this steps at next update, thanks for help review.